src/Database/Query.php

```diff
@@ -335,7 +337,11 @@ public function traverse(Closure $callback)
      */
     public function traverseParts(Closure $visitor, array $parts)
     {
+        $debugMode = Configure::read('debug');
```

Comment:
maybe I am mistaken but do the parts not define what SQL is actually being executed at the end in the driver?
Why execute SQL queries with comments in them on production?
Sure comments may be good when debugging queries but I personally wouldn't want comments being part of the queries which are executed on prod.

Reply:
I am also fine with the idea to add a flag to the connection so its configurable instead of checking the debug state.

Reply:
>  Why execute SQL queries with comments in them on production?

Because that's what the developer asked for. If users don't want comments in their queries they could not add them.

Reply:
If we define it that way I am fine with that as well.

Reply:
I think having the comment show up when set is the most transparent and simple interface we can provide.